network/kademlia.go

```diff
@@ -1061,6 +1069,19 @@ func (k *Kademlia) connectedNeighbours(peers [][]byte) (got bool, n int, missing
 	return gots == len(peers), gots, culprits
 }
 
+//calculates the expected min size of a given bin (minBinSize)
+//binSize = k.MinBinSize * 2 ^ (depth - binNumber - 1)
+func (k *Kademlia) expectedMinBinSize(binNumber int) int {
+	//return 2
+	depth := depthForPot(k.conns, k.NeighbourhoodSize, k.base)
+
+	minBinSize := k.MinBinSize * int(math.Pow(2, float64(depth-binNumber-1)))
```

Comment:
Perhaps I recall incorrectly, but didn't we agree on a linear progression here to start out with? Eg. if depth is 3 po 2 would have 2, po 1 would have 3, po 0 would have 4?

Reply:
I don't recall exactly, but we were talking about making it proportional to the number of requests per bin. Although with a linear progression bins with more request will have more peers, this number of peers will not be proportional to the number of requests. Do you feel like a linear progression would be a better starting point ?

Reply:
Yes, I do.

In fact, I realize that we haven't really considered what _total_ amount of requests a peer will be getting from the whole network depending on the cardinalities of bins. To look at number of requests to a peer merely from the perspective of a single node is probably not very useful. How to properly test these properties remains an open question for now, but it should be discussed and a plan should be laid for making such tests.

Reply:
https://swarmresear.ch/t/light-client-connection-strategy/17/3
I was taking a look at  this and, although we don't have yet the capabilities for differentiating between light and normal connections, it seems like we would only want this PR's new suggestPeer when making light connections? and just have a fixed number of peers per bin for full connections. 
For light connections, the nodes per bin progression would be in line with the one proposed in this PR. 
I agree that having a way of testing the different approaches is needed before deciding on something

Reply:
Hmm. Right. So it's not as simple as all that after all. According to the reasoning in this article we would have to unset the syncing capability with the peers above the `MinBinSize` limit, and reset it when one of the syncing peers disconnect (and stay disconnected for awhile). In fact, we have the same issue when we get explicit incoming connections (which aren't rejected) after bin is saturated.

Perhaps a SWIP should be written here to suggest an implementation of these distinctions could look like.

As we don't even have working capabilities for the syncing, _nor_ the current capability to change ... capabilities on the fly, I would vote that we can merge this PR with linear progession of number of peers and improve from there.

Reply:
Ok, sounds good, will change it